src/app/zone-ingresses/sources.ts

```diff
@@ -0,0 +1,78 @@
+import { DataSourceResponse } from '@/app/application/services/data-source/DataSourcePool'
+import type KumaApi from '@/services/kuma-api/KumaApi'
+import type { PaginatedApiListResponse as CollectionResponse } from '@/types/api.d'
+import type { ZoneIngressOverview, ZoneIngress } from '@/types/index.d'
+
+type PaginationParams = {
+  size: number
+  page: number
+}
+
+type DetailParams = {
+  name: string
+}
+
+type EnvoyDataParams = DetailParams & {
+  dataPath: 'xds' | 'clusters' | 'stats'
+}
+
+type Closeable = { close: () => void }
+
+export type ZoneIngressSource = DataSourceResponse<ZoneIngress>
+export type ZoneIngressOverviewCollection = CollectionResponse<ZoneIngressOverview>
+export type ZoneIngressOverviewSource = DataSourceResponse<ZoneIngressOverview>
+export type ZoneIngressOverviewCollectionSource = DataSourceResponse<ZoneIngressOverviewCollection>
+
+export type EnvoyDataSource = DataSourceResponse<object | string>
+
+export const sources = (api: KumaApi) => {
+  return {
+
+    '/zone-cps/:name/ingresses': async (params: DetailParams & PaginationParams, source: Closeable) => {
+      source.close()
+
+      const { name, size, page } = params
+      const offset = size * (page - 1)
+
+      const res = await api.getAllZoneIngressOverviews({ size, offset })
+      res.items = res.items.filter((item) => {
+        return item.zoneIngress.zone === name
+      })
+      res.total = res.items.length
```

Comment:
Frontend filtering again, but no `*` this time as we never need "gimme all of them" for ingresses.

Reply:
Same question here: do we have a ticket for filtering this on the API?

Reply:
See above